New option to allow for 'Ignored' migrations in 'validate' command: ignoreIgnoredMigrations

bkaminnski opened the pull request:
This is a pull request for issue #1865. 

The main change was done in MigrationInfoImpl class, where the new configuration flag is actually checked. All other changes are done taking as an example "flyway.ignoreMissingMigrations" configuration flag. I added a new option: "flyway.ignoreIgnoredMigrations". All changes in code take place next to a "missing" boolean flag.

I tried to follow all the conventions already applied in code. Also description of the configuration option in flyway.conf was done so that it is as similar to the description of surrounding "ignore..." flags, as possible.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/flyway/flyway?pullRequest=1866) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/flyway/flyway?pullRequest=1866) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/flyway/flyway?pullRequest=1866) it.</sub>

axelfontaine commented:
Thanks. This looks great! Could you also make the necessary changes to the Maven and Gradle plugins?

bkaminnski commented:
Done. Sorry I missed that. I also added info message to command-line help and checked that it now shows up. I haven't run flyway as maven nor gradle plugin though. Could you please let me know whether you have a quick way to verify that? I can do this, but I need little more time to set up maven and gradle plugin in my environment to make sure that this new flag is properly picked up also here.

axelfontaine commented:
Looking good!

All that is left now is the docs. Could you submit a pull request against the flyway-5.1.0 branch of flyway/flywaydb.org ? Thanks!

bkaminnski commented:
Done. What I noticed during update of documentation:

1. I missed handling of FLYWAY_IGNORE_IGNORED_MIGRATIONS in env variables in ConfigUtils. Fixed in another commit in pull request.
1. I updated 3 undo.md files (commandline, gradle and maven) - this is however not covered by opensource version, and I am not sure how this relates to my change.
1. I skipped release notes and javadoc.

axelfontaine commented:
Thank you very much for the great work! Merged. I have also taken the liberty to add you to the Hall of Fame page (flyway-5.1.0 branch).
